akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -663,10 +663,13 @@ private[stream] final class GraphInterpreter(
   private[stream] def fail(connection: Int, ex: Throwable): Unit = {
     val currentState = portStates(connection)
     if (Debug) println(s"$Name   fail($connection, $ex) [$currentState]")
-    portStates(connection) = currentState | (OutClosed | InFailed)
-    if ((currentState & InClosed) == 0) {
+    portStates(connection) = currentState | OutClosed
+    if ((currentState & (InClosed | OutClosed)) == 0) {
```

Comment:
This was the actual issue. We overloaded the internal queue since we tried to enqueue after the port has been already closed.